pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -80,12 +81,17 @@ func newProxierHealthServer(listener listener, httpServerFactory httpServerFacto
 
 // Updated updates the lastUpdated timestamp.
 func (hs *proxierHealthServer) Updated() {
+	hs.oldestPendingQueued.Store(zeroTime)
 	hs.lastUpdated.Store(hs.clock.Now())
 }
 
 // QueuedUpdate updates the lastQueued timestamp.
 func (hs *proxierHealthServer) QueuedUpdate() {
-	hs.lastQueued.Store(hs.clock.Now())
+	if val := hs.oldestPendingQueued.Load(); val != nil {
+		if val.(time.Time).IsZero() {
+			hs.oldestPendingQueued.Store(hs.clock.Now())
+		}
+	}
```

Comment:
oh, sorry, I should have caught this before; this needs to be an atomic operation. Which actually makes it simpler anyway.

```suggestion
	// Set oldestPendingQueued only if it's currently zero
	hs.oldestPendingQueued.CompareAndSwap(zeroTime, hs.clock.Now())
```

Reply:
Done